x-pack/plugins/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -248,6 +256,12 @@ export class AlertsClient<
   }
 
   public async persistAlerts() {
+    if (!this.ruleType.alerts?.shouldWrite) {
```

Comment:
I think we need to add this check to the `initializeExecution` function in the alerts client as well. Previously when `shouldWrite = false`, we only instantiated the LegacyAlertsClient. The AlertsClient wraps the LegacyAlertsClient but also reads from the alerts index at the start of rule execution and writes to the alerts index at the end of rule execution. This check here skips writing to the alerts index at the end of rule execution and we should also skip reading from the alerts index at the start of rule execution for now because we're still using the lifecycle & persistence executors which do the reading for us.

Right now, `initializeExecution` looks like:

```
  public async initializeExecution(opts: InitializeExecutionOpts) {
    await this.legacyAlertsClient.initializeExecution(opts);

    // Get tracked alert UUIDs to query for
    // TODO - we can consider refactoring to store the previous execution UUID and query
    // for active and recovered alerts from the previous execution using that UUID
    const trackedAlerts = this.legacyAlertsClient.getTrackedAlerts();

    const uuidsToFetch: string[] = [];
    keys(trackedAlerts).forEach((key) => {
      const tkey = key as keyof TrackedAlerts<LegacyState, LegacyContext>;
      keys(trackedAlerts[tkey]).forEach((alertId: string) => {
        uuidsToFetch.push(trackedAlerts[tkey][alertId].getUuid());
      });
    });
...
}
```

We should do:

```
  public async initializeExecution(opts: InitializeExecutionOpts) {
    await this.legacyAlertsClient.initializeExecution(opts);
    if (!this.ruleType.alerts?.shouldWrite) {
      return;
    }

    // Get tracked alert UUIDs to query for
    // TODO - we can consider refactoring to store the previous execution UUID and query
    // for active and recovered alerts from the previous execution using that UUID
    const trackedAlerts = this.legacyAlertsClient.getTrackedAlerts();

    const uuidsToFetch: string[] = [];
    keys(trackedAlerts).forEach((key) => {
      const tkey = key as keyof TrackedAlerts<LegacyState, LegacyContext>;
      keys(trackedAlerts[tkey]).forEach((alertId: string) => {
        uuidsToFetch.push(trackedAlerts[tkey][alertId].getUuid());
      });
    });
```